modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -216,49 +250,173 @@ struct GOCVBGR: public cv::detail::KernelTag
     static cv::gapi::streaming::GStreamingKernel kernel() { return {&create}; };
 };
 
-void GOCVBGR::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
-                     cv::gimpl::GIslandExecutable::IOutput &out)
+void GOCVBGR::Actor::extractRMat(const cv::MediaFrame& frame, cv::RMat& rmat)
 {
-    const auto in_msg = in.get();
-    if (cv::util::holds_alternative<cv::gimpl::EndOfStream>(in_msg))
+    const auto& desc = frame.desc();
+    switch (desc.fmt)
     {
-        out.post(cv::gimpl::EndOfStream{});
-        return;
+        case cv::MediaFormat::BGR:
+        {
+            rmat = cv::make_rmat<cv::gimpl::RMatMediaFrameAdapter>(frame,
+            [](const cv::GFrameDesc& d){ return cv::GMatDesc(CV_8U, 3, d.size); },
+            [](const cv::GFrameDesc& d, const cv::MediaFrame::View& v){
+                return cv::Mat(d.size, CV_8UC3, v.ptr[0], v.stride[0]);
+            });
+            break;
+        }
+        case cv::MediaFormat::NV12:
+        {
+            std::call_once(m_warnFlag,
+                [](){
+                    GAPI_LOG_WARNING(NULL, "\nOn-the-fly conversion from NV12 to BGR will happen.\n"
+                        "Conversion may cost a lot for images with high resolution.\n"
+                        "To retrieve cv::Mat-s from NV12 cv::MediaFrame for free, you may use "
+                        "cv::gapi::streaming::Y and cv::gapi::streaming::UV accessors.\n");
```

Comment:
This may be misleading too since `Y` and `UV` are working at `cv::GMat` level (I mean, for user), there's no Mats etc.